src/main/java/com/limechain/trie/structure/TrieStructure.java

```diff
@@ -495,4 +496,52 @@ public <U> boolean structurallyEquals(TrieStructure<U> other) {
             }
         }
     }
+
+    public boolean removeNode(Nibbles key) {
+        return switch (existingNodeInner(key)) {
+            case ExistingNodeInnerResult.NotFound ignored -> false;
+            case ExistingNodeInnerResult.Found found -> removeNode(found.nodeIndex);
+        };
+    }
+
+    public boolean removeNode(int nodeIndex) {
```

Comment:
I would change this `nodeIndex` argument's type from `int` to `TrieNodeIndex`, since it's part of the public API.
As far as I could gather, it's singular usage outside the package allows for it (as was intended).